pkg/prometheus/promcfg.go

```diff
@@ -3511,6 +3511,59 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// HetznerSDConfig
+	if len(sc.Spec.HetznerSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.HetznerSDConfigs))
+		for i, config := range sc.Spec.HetznerSDConfigs {
+			assetStoreKey := fmt.Sprintf("scrapeconfig/%s/%s/hetznersdconfig/%d", sc.GetNamespace(), sc.GetName(), i)
+			configs[i] = append(configs[i], yaml.MapItem{
+				Key:   "role",
```

Comment:
nit: Lets move appending the role after the auth related sections to avoid clubbing different things